zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+# SPDX-License-Identifier: MIT
+
+if(CONFIG_LIBOQS)
+        # We add our own module library to build our RNG implementation
+        zephyr_library()
+        zephyr_library_sources(${ZEPHYR_CURRENT_MODULE_DIR}/zephyr/zephyr_rng.c)
+
+        # Configuration for liboqs
+        set(OQS_DIST_BUILD  OFF)
+        set(OQS_BUILD_ONLY_LIB ON)
+        set(OQS_USE_OPENSSL OFF)
+        set(OQS_POSITION_INDEPENDENT_CODE OFF)
+        set(OQS_USE_EXTERNAL_RNG ON)
+
+        set(CMAKE_CROSSCOMPILING ON)
+        set(CMAKE_SIZEOF_VOID_P 4) # Not really needed, only for surpressing unrelated warnings
+
+        # Disable features by hand, as CMake won't find them properly with Zephyr
+        set(CMAKE_HAVE_GETENTROPY OFF)
+        set(CMAKE_HAVE_ALIGNED_ALLOC OFF)
+        set(CMAKE_HAVE_POSIX_MEMALIGN OFF)
+        set(CMAKE_HAVE_MEMALIGN OFF)
+        set(CMAKE_HAVE_EXPLICIT_BZERO OFF)
+        set(CMAKE_HAVE_MEMSET_S OFF)
+        set(CC_SUPPORTS_WA_NOEXECSTACK OFF)
+        set(LD_SUPPORTS_WL_Z_NOEXECSTACK OFF)
+
+        # Algorithm selection
+        if(CONFIG_LIBOQS_ENABLE_KEM_BIKE)
```

Comment:
Yes, those are also set by the Kconfig system from within Zehpyr. This is the only way to configure the build for Zephyr. Hence, we cannot use the OQS defines directly, but have to „proxy“ them in this „ugly“ way.

Reply:
Understood. That makes me wonder how (whether) we shall test this integration: Shall this be done within `liboqs` CI (which would mean we somehow have to set these configs) or do you want to rely on doing this (only) within Zephyr CI?

Reply:
After looking at the test infrastructure, I think it makes sense to test the Zephyr port within the `liboqs` CI. There is a docker container available from Zephyr with the SDK ready to go, which should make an integration relatively easy. 

I thought about mirroring the tests from `test_kem.c` and `test_sig.c`, one time with only the NIST selected algorithms enabled (and testing for the remaining ones to be disabled, hence, testing the ENABLE options above), and one time with all algorithms enabled. 

Do you think other tests are necessary, too? Unfortunately, I think won't have time for the tests until the end of this week, but I will add them if you confirm.

Reply:
> and one time with all algorithms enabled.

I'd _only_ go with this as the NIST algs are just a subset. But then again, let's see what the runtime of this will be: If it's very long, it may be better to scale things back...

> I think won't have time for the tests until the end of this week, but I will add them if you confirm.

No rush here. Please do at a pace that suits your other activities.

Reply:
I added two sample applications in the `zephyr` directories for KEMs and Signatures. Both are copied and adapted from `test_kem.c` and `test_sig.c` to Zephyr. These examples are now also used in a new Github CI Action to test the two applications (execute them). I added Zephyr Version 3.4 and 3.5.

I also took a look at the CircleCI, but I even can't get those tests running locally on my side. Can you give me any hints on that side or are the Github Actions enough?

Reply:
Thanks for these tests!

> are the Github Actions enough

Absolutely; we actually prefer Github over CCI.